service/tpo/tpo.go

```diff
@@ -85,20 +85,24 @@ func New(config Config) (*Service, error) {
 	return eventer, nil
 }
 
-func (s *Service) Ensure(tpo draughtsmantpr.CustomObject) error {
-	if tpo.TypeMeta.APIVersion == "" {
-		tpo.TypeMeta.APIVersion = s.draughtsmanTPR.APIVersion()
+func (s *Service) Ensure(TPO *draughtsmantpr.CustomObject) error {
+	if TPO.TypeMeta.APIVersion == "" {
+		TPO.TypeMeta.APIVersion = s.draughtsmanTPR.APIVersion()
 	}
-	if tpo.TypeMeta.Kind == "" {
-		tpo.TypeMeta.Kind = s.draughtsmanTPR.Kind()
+	if TPO.TypeMeta.Kind == "" {
+		TPO.TypeMeta.Kind = s.draughtsmanTPR.Kind()
+	}
+	if TPO.ObjectMeta.Name == "" {
+		TPO.ObjectMeta.Name = Name
 	}
 
-	endpoint := s.draughtsmanTPR.Endpoint(DefaultNamespace) + "/" + Name
-	_, err := s.k8sClient.Core().RESTClient().Post().Body(tpo).AbsPath(endpoint).DoRaw()
+	endpoint := s.draughtsmanTPR.Endpoint(DefaultNamespace)
```

Comment:
Here are all changes to make the TPO creation work. One important thing is to post against the collection, not the resource itself.